libr/anal/esil_plugin.c

```diff
@@ -0,0 +1,80 @@
+/* radare2 - LGPL - Copyright 2021 - condret */
+
+#include <r_anal.h>
+#include <r_list.h>
+#include <config.h>
+#include "../config.h"
+
+static RAnalEsilPlugin *esil_static_plugins[] = {
+	R_ESIL_STATIC_PLUGINS
+};
+
+R_API void r_anal_esil_plugins_init(RAnalEsil *esil) {
+	r_return_if_fail (esil);
+	esil->plugins = r_list_new ();
+	esil->active_plugins = r_list_new ();
+	ut32 i = 0;
+	while (esil_static_plugins[i]) {
+		r_anal_esil_plugin_add (esil, esil_static_plugins[i]);
+		i++;
+	}
+}
+
+R_API void r_anal_esil_plugins_fini(RAnalEsil *esil) {
+	r_return_if_fail (esil && esil->plugins && esil->active_plugins);
+	while (r_list_length (esil->active_plugins)) {
+		RAnalEsilActivePlugin *eap = (RAnalEsilActivePlugin *)r_list_pop (esil->active_plugins);
+		eap->plugin->fini (esil, eap->user);
+		free (eap);
+	}
+	r_list_free (esil->active_plugins);
+	r_list_free (esil->plugins);
+}
+
+R_API bool r_anal_esil_plugin_add(RAnalEsil *esil, RAnalEsilPlugin *plugin) {
+	r_return_val_if_fail (esil && esil->plugins && plugin &&
+				plugin->init && plugin->fini, false);
+	r_list_append (esil->plugins, plugin);
+	return true;
+}
+
+R_API bool r_anal_esil_plugin_activate(RAnalEsil *esil, const char *name) {
+	r_return_val_if_fail (esil && esil->plugins &&
+				esil->active_plugins && name, false);
+	RListIter *iter;
+	RAnalEsilActivePlugin *eap;
+	r_list_foreach (esil->active_plugins, iter, eap) {
+		// check if plugin is already activated
+		if (!strcmp (eap->plugin->name, name)) {
+			return false;	// is this correct?
+		}
+	}
+	RAnalEsilPlugin *ep;
+	r_list_foreach (esil->plugins, iter, ep) {
+		if (!strcmp(ep->name, name)) {
```

Comment:
better use a hashtable to get the plugin by name instead of iterating them all everytime, this seems to be avery common operation